### PR TITLE
[NEWDEV] Disable NewFolder btn and Show browser title when selecting a directory

### PR DESCRIPTION
**Pull request by zecarlos1957**
[NEWDEV] Show browser title when selecting a directory

Disable the new folder button

### CODE REVIEW

File: dll/win32/newdev/lang/zh-TW.rc [hunk 1]
```diff
@@ -95,5 +95,8 @@ BEGIN
     IDS_CHSOURCE_TITLE "選擇搜索和安裝選項"
     IDS_SEARCHDRV_TITLE "正在搜索..."
     IDS_INSTALLDRV_TITLE "正在安裝..."
+
+    /* For  IDD_BROWSE_FOR_FOLDER */
```
Comment (binarymaster) [inline dll/win32/newdev/lang/zh-TW.rc:99]: Extra space here and in other places:
```suggestion
    /* For IDD_BROWSE_FOR_FOLDER */
```

File: dll/win32/newdev/wizard.c [hunk 2]
```diff
@@ -641,9 +641,13 @@ CHSourceDlgProc(
                 {
                     BROWSEINFO bi = { 0 };
                     LPITEMIDLIST pidl;
+                    HINSTANCE hInst = (HINSTANCE)GetWindowLong(hwndDlg, GWL_HINSTANCE);
+                    WCHAR Title[MAX_PATH];
+                    LoadStringW(hInst, IDS_BROWSE_FOR_FOLDER_TITLE, Title, MAX_PATH);
```
Comment (binarymaster) [inline dll/win32/newdev/wizard.c:646]: Perhaps you can replace hardcoded `MAX_PATH` with `_countof` here:
```suggestion
                    LoadStringW(hInst, IDS_BROWSE_FOR_FOLDER_TITLE, Title, _countof(Title));
```

File: dll/win32/newdev/wizard.c [hunk 3]
```diff
@@ -641,9 +641,13 @@ CHSourceDlgProc(
                 {
                     BROWSEINFO bi = { 0 };
                     LPITEMIDLIST pidl;
+                    HINSTANCE hInst = (HINSTANCE)GetWindowLong(hwndDlg, GWL_HINSTANCE);
```
Comment (JoachimHenze) [inline dll/win32/newdev/wizard.c:644]: That should be GetWindowLongPtr()! (And maybe GWLP_ ? also)

**Review by JoachimHenze — CHANGES_REQUESTED** [review]

File: dll/win32/newdev/wizard.c (hunk 3 above)
Comment (zecarlos1957) [reply dll/win32/newdev/wizard.c]: Thanks

File: dll/win32/newdev/lang/pl-PL.rc [hunk 4]
```diff
@@ -101,4 +101,7 @@ BEGIN
     IDS_CHSOURCE_TITLE "Wybierz opcje wyszukiwania i instalacji"
     IDS_SEARCHDRV_TITLE "Szukanie..."
     IDS_INSTALLDRV_TITLE "Instalowanie..."
+
+    /* For IDD_BROWSE_FOR_FOLDER */
+    IDS_BROWSE_FOR_FOLDER_TITLE "Select the folder containing the drivers for the hardware."
```
Comment (Saibamen) [inline dll/win32/newdev/lang/pl-PL.rc:106]:
```suggestion
    IDS_BROWSE_FOR_FOLDER_TITLE "Wybierz folder zawierający sterowniki dla sprzętu."
```

@CC: @reactos/lang-polish

Comment (zecarlos1957) [reply dll/win32/newdev/lang/pl-PL.rc:106]: After this PR is merged, you can open a new PULL REQUEST with the translation in your language. 
I don't know Polish

File: dll/win32/newdev/wizard.c (hunk 3 above)
Comment (zecarlos1957) [reply dll/win32/newdev/wizard.c]: @tkreuzer  can you confirm this?

File: dll/win32/newdev/wizard.c (hunk 2 above)
Comment (binarymaster) [reply dll/win32/newdev/wizard.c:646]: You didn't applied this change, and probably resolved it by accident.

Comment (zecarlos1957) [reply dll/win32/newdev/wizard.c]: OPPS. Thanks

**Review by binarymaster — APPROVED** [review]
LGTM. Going to merge as soon as our buildbots are back.

Also I'll probably move pl-PL translation to PR #4289 right before merging.

File: dll/win32/newdev/lang/bg-BG.rc [hunk 5]
```diff
@@ -93,4 +93,7 @@ BEGIN
     IDS_CHSOURCE_TITLE "Choose search and install options"
     IDS_SEARCHDRV_TITLE "Searching..."
     IDS_INSTALLDRV_TITLE "Installing..."
+
+    /* For IDD_BROWSE_FOR_FOLDER */
+    IDS_BROWSE_FOR_FOLDER_TITLE "Select the folder containing the drivers for the hardware."
```
Comment (HBelusca) [inline dll/win32/newdev/lang/bg-BG.rc:98]: Better sounding english translation (thanks jeffbox & TheDarkFire):
```suggestion
    IDS_BROWSE_FOR_FOLDER_TITLE "Select the folder containing proper drivers for this hardware."
```

File: dll/win32/newdev/wizard.c [hunk 6]
```diff
@@ -641,9 +641,13 @@ CHSourceDlgProc(
                 {
                     BROWSEINFO bi = { 0 };
                     LPITEMIDLIST pidl;
+                    HINSTANCE hInst = (HINSTANCE)GetWindowLongPtr(hwndDlg, GWLP_HINSTANCE);
```
Comment (HBelusca) [inline dll/win32/newdev/wizard.c:644]: There seems to be already the `hDllInstance` global available for this.

File: dll/win32/newdev/wizard.c [hunk 7]
```diff
@@ -641,9 +641,13 @@ CHSourceDlgProc(
...
+                    WCHAR Title[MAX_PATH];
+                    LoadStringW(hInst, IDS_BROWSE_FOR_FOLDER_TITLE, Title, _countof(Title));
```
Comment (binarymaster) [inline dll/win32/newdev/wizard.c:646]: Based on @HBelusca suggestion:
```suggestion
                    WCHAR Title[MAX_PATH];
                    LoadStringW(hDllInstance, IDS_BROWSE_FOR_FOLDER_TITLE, Title, _countof(Title));
```

File: dll/win32/newdev/lang/bg-BG.rc (hunk 5 above)
Comment (zecarlos1957) [reply dll/win32/newdev/lang/bg-BG.rc]: Thanks a lot

File: dll/win32/newdev/lang/pl-PL.rc [hunk 8]
```diff
@@ -101,4 +101,7 @@ BEGIN
     IDS_CHSOURCE_TITLE "Wybierz opcje wyszukiwania i instalacji"
     IDS_SEARCHDRV_TITLE "Szukanie..."
     IDS_INSTALLDRV_TITLE "Instalowanie..."
+
+    /* For IDD_BROWSE_FOR_FOLDER */
+    IDS_BROWSE_FOR_FOLDER_TITLE "Wybierz folder zawierający sterowniki dla sprzętu."
```
Comment (Saibamen) [inline dll/win32/newdev/lang/pl-PL.rc:106]:
```suggestion
    IDS_BROWSE_FOR_FOLDER_TITLE "Wybierz folder zawierający odpowiednie sterowniki dla tego sprzętu."
```

Update after https://github.com/reactos/reactos/pull/4288/commits/92fcbf5b537810e30f49a2bf15adc2e80660621d

File: dll/win32/newdev/lang/pt-PT.rc [hunk 9]
```diff
@@ -98,4 +98,7 @@ BEGIN
     IDS_CHSOURCE_TITLE "Escolha as opções de pesquisa e instalação"
     IDS_SEARCHDRV_TITLE "A procurar..."
     IDS_INSTALLDRV_TITLE "A instalar..."
+
+    /* For IDD_BROWSE_FOR_FOLDER */
+    IDS_BROWSE_FOR_FOLDER_TITLE "Seleccione a pasta que contém controladores para o hardware."
```
Comment (binarymaster) [inline dll/win32/newdev/lang/pt-PT.rc:103]: You probably want to update pt-PT translation (reflecting en-US change)
```suggestion
    IDS_BROWSE_FOR_FOLDER_TITLE "Seleccione a pasta que contém controladores adequados para este hardware."
```

File: dll/win32/newdev/lang/bg-BG.rc [hunk 10]
```diff
@@ -93,4 +93,7 @@ BEGIN
     IDS_CHSOURCE_TITLE "Choose search and install options"
     IDS_SEARCHDRV_TITLE "Searching..."
     IDS_INSTALLDRV_TITLE "Installing..."
+
+    /* For IDD_BROWSE_FOR_FOLDER */
```
Comment (JoachimHenze) [inline dll/win32/newdev/lang/bg-BG.rc:97]: remove that comment and the empty line above for all languages. The title is absolutely self-description-like.
If everyone did it like that, the rc files would look rather ugly.
Same for all other languages!

File: dll/win32/newdev/lang/de-DE.rc [hunk 11]
```diff
@@ -93,4 +93,7 @@ BEGIN
     IDS_CHSOURCE_TITLE "Wählen Sie Such- und Installationsoptionen"
     IDS_SEARCHDRV_TITLE "Suche..."
     IDS_INSTALLDRV_TITLE "Installiere..."
+
+    /* For IDD_BROWSE_FOR_FOLDER */
+    IDS_BROWSE_FOR_FOLDER_TITLE "Select the folder containing proper drivers for this hardware."
```
Comment (JoachimHenze) [inline dll/win32/newdev/lang/de-DE.rc:98]: "Wählen Sie den Ordner aus, der den korrekten Treiber für dieses Gerät enthält."
is the de-DE Translation.

**Review by JoachimHenze — CHANGES_REQUESTED** [review]

File: dll/win32/newdev/lang/de-DE.rc (hunk 11 above)
Comment (binarymaster) [reply dll/win32/newdev/lang/de-DE.rc:98]: @JoachimHenze I'm going to revert all translations that do not belong to PR author anyway before merge, please do this as separate PR / commit once it's merged.

**Comment by binarymaster** [comment]
Ok it seems @gonzoMD has broke my plans a bit by merging PR #4289 where I actually wanted to add PL translation from Adam...

So I will merge all translations here.

**Review by JoachimHenze — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]
Everything is addressed now.

**Comment by gonzoMD** [comment]
> Ok it seems @gonzoMD has broke my plans a bit by merging PR #4289 where I actually wanted to add PL translation from Adam...
> 
> So I will merge all translations here.
> 
> 

Ooopsie

File: dll/win32/newdev/lang/fr-FR.rc [hunk 12]
```diff
@@ -93,4 +93,5 @@ BEGIN
     IDS_CHSOURCE_TITLE "Options de recherche et d'installation"
     IDS_SEARCHDRV_TITLE "Recherche..."
     IDS_INSTALLDRV_TITLE "Installation..."
+    IDS_BROWSE_FOR_FOLDER_TITLE "Select the folder containing proper drivers for this hardware."
```
Comment (HBelusca) [inline dll/win32/newdev/lang/fr-FR.rc:96]:
```suggestion
    IDS_BROWSE_FOR_FOLDER_TITLE "Veuillez sélectionner le dossier contenant les pilotes correspondant à ce matériel."
```

**Review by HBelusca — APPROVED** [review]

File: dll/win32/newdev/wizard.c (hunk 3 above)
Comment (HBelusca) [reply dll/win32/newdev/wizard.c]: This is correct (using the "Ptr" version) for portability of the code between 32 and 64 bits.

**Pull request merged**